### PR TITLE
[docs] Add sources policy

### PR DESCRIPTION
**Pull request by danimtb**
Closes https://github.com/conan-io/conan-center-index/issues/3296

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * User documentation
   + [Contributing to Conan Center Index](contributing.md)
   + [Adding Packages to ConanCenter](how_to_add_packages.md)
+  + [Packaging policy](packaing_policy.md)
```
Comment (intelligide) [inline docs/README.md:11]:
```suggestion
  + [Packaging policy](packaging_policy.md)
```

File: docs/packaging_policy.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+- Recipes should always build packages from library sources.
+
+- Library sources should come from an official origin like the library source code repository or the official release/download webpage.
+  Source forks or unofficial origin of sources are not allowed.
+
+- If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their format, an issue should be
+  opened in order to discuss alternative solutions.
+
+- Library sources that are no publicly available will not be allowed in this repository even if the license allows their redistribution.
```
Comment (madebr) [inline docs/packaging_policy.md:16]:
```suggestion
- Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.
```

File: docs/packaging_policy.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+- Recipes should always build packages from library sources.
+
+- Library sources should come from an official origin like the library source code repository or the official release/download webpage.
+  Source forks or unofficial origin of sources are not allowed.
+
+- If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their format, an issue should be
+  opened in order to discuss alternative solutions.
+
+- Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.
```
Comment (jgsogo) [inline docs/packaging_policy.md:16]: IMO these two points are the same: sources that are not accessible. 

We are not storing sources. The recommendation is to contact the publisher and ask them to provide the sources in a way/format that can be consumed programmatically.

File: docs/packaging_policy.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+- In the remote case of having to package precompiled libraries, those should come from official sources like release/download webpages.
```
Comment (jgsogo) [inline docs/packaging_policy.md:18]: To add here (maybe reformulate together with the previous point): if binaries are not accessible the CTA is to contact the publisher and ask for a programmatically way to retrieve and consume those binaries.

If these binaries are needed as `build_requires` for some library, then we can write a `<app>/system` recipe and consider making those binaries available in the CI machines (if license let us install them) (#2850)

**Review by jgsogo — CHANGES_REQUESTED** [review]
Right now the policy is that we are not serving sources/binaries that original publishers are not serving themselves. I think this should be clear in this document.

If there is enough pressure to modify this policy we can escalate this petition and it can be reconsidered. So far, I think it is only #2850 the one being affected by this.

**Comment by madebr** [comment]
What would be useful imho, is adding recipes for public sources/binaries, but that do not allow redistribution.
This would allow packaging binaries of Microsoft, which almost always forbid redistribution by some EULA.

c3i can check whether the recipe is correct, but no packages are uploaded to bintray.

**Comment by jgsogo** [comment]
> What would be useful imho, is adding recipes for public sources/binaries, but that do not allow redistribution.
> This would allow packaging binaries of Microsoft, which almost always forbid redistribution by some EULA.
> 
> c3i can check whether the recipe is correct, but no packages are uploaded to bintray.

I've thought about it and maybe it makes sense to have a new type of recipe that retrieves the sources/binaries only on the consumer machine. A recipe with [`build_policy = always`](https://docs.conan.io/en/latest/reference/conanfile/attributes.html#build-policy)... and ConanCenter smart enough to not uploaded packages (but run the tests) for them.

Probably this deserves a new issue, and maybe some alignment with future Conan v2 to know if `build_policy` is the best approach for it or there will be something better.

File: docs/packaging_policy.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Packaging Policy
+
+This document gathers all the relevant information regarding the general lines to follow while creating new recipes that will eventually be
+part of this repository.
+
+## Sources
+
+- Recipes should always build packages from library sources.
```
Comment (prince-chrismc) [inline docs/packaging_policy.md:8]: The spec writter in me :cursing_face: "should always"

```suggestion
- Recipes should build packages from library sources.
```

**Comment by danimtb** [comment]
I have updated the policy with some suggestions and clarified the points under sections. I would like to move this forward as we really need to set these points in order to clarify the status of some pull requests. cc/ @SSE4 @uilianries

**Review by prince-chrismc — APPROVED** [review]

**Review by madebr — APPROVED** [review]

File: docs/packaging_policy.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+**Sources not accessible:**
+
+- Library sources that are not publicly available will not be allowed in this repository even if the license allows their redistribution.
+
+- If library sources cannot be downloaded from their official origin or cannot be consumed directly due to their
+  format, the recommendation is to contact the publisher and ask them to provide the sources in a way/format that can be consumed
+  programmatically.
+
+- In case of needing those binaries to use them as a "build require" for some library, we will consider following the approach of adding it
+  as a system recipe (`<build_require>/system``) and making those binaries available in the CI machines (if the license allows it).
```
Comment (SSE4) [inline docs/packaging_policy.md:22]:
```suggestion
  as a system recipe (`<build_require>/system`) and making those binaries available in the CI machines (if the license allows it).
```

**Review by SSE4 — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**